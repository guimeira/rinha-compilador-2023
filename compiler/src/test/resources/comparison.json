{"name":"comparison.rinha","expression":{"kind":"Let","name":{"text":"num","location":{"start":4,"end":7,"filename":"comparison.rinha"}},"value":{"kind":"Int","value":5,"location":{"start":10,"end":11,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"b","location":{"start":17,"end":18,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":21,"end":24,"filename":"comparison.rinha"}},"op":"Gt","rhs":{"kind":"Int","value":10,"location":{"start":27,"end":29,"filename":"comparison.rinha"}},"location":{"start":21,"end":29,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":35,"end":36,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"b","location":{"start":45,"end":46,"filename":"comparison.rinha"}},"location":{"start":39,"end":47,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"b","location":{"start":53,"end":54,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Int","value":10,"location":{"start":57,"end":59,"filename":"comparison.rinha"}},"op":"Gt","rhs":{"kind":"Var","text":"num","location":{"start":62,"end":65,"filename":"comparison.rinha"}},"location":{"start":57,"end":65,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":71,"end":72,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"b","location":{"start":81,"end":82,"filename":"comparison.rinha"}},"location":{"start":75,"end":83,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"c","location":{"start":89,"end":90,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":93,"end":96,"filename":"comparison.rinha"}},"op":"Lt","rhs":{"kind":"Int","value":10,"location":{"start":99,"end":101,"filename":"comparison.rinha"}},"location":{"start":93,"end":101,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":107,"end":108,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"c","location":{"start":117,"end":118,"filename":"comparison.rinha"}},"location":{"start":111,"end":119,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"d","location":{"start":125,"end":126,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":129,"end":132,"filename":"comparison.rinha"}},"op":"Eq","rhs":{"kind":"Int","value":10,"location":{"start":136,"end":138,"filename":"comparison.rinha"}},"location":{"start":129,"end":138,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":144,"end":145,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"d","location":{"start":154,"end":155,"filename":"comparison.rinha"}},"location":{"start":148,"end":156,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"e","location":{"start":162,"end":163,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":166,"end":169,"filename":"comparison.rinha"}},"op":"Eq","rhs":{"kind":"Int","value":5,"location":{"start":173,"end":174,"filename":"comparison.rinha"}},"location":{"start":166,"end":174,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":180,"end":181,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"e","location":{"start":190,"end":191,"filename":"comparison.rinha"}},"location":{"start":184,"end":192,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"f","location":{"start":198,"end":199,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":202,"end":205,"filename":"comparison.rinha"}},"op":"Gte","rhs":{"kind":"Int","value":5,"location":{"start":209,"end":210,"filename":"comparison.rinha"}},"location":{"start":202,"end":210,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":216,"end":217,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"f","location":{"start":226,"end":227,"filename":"comparison.rinha"}},"location":{"start":220,"end":228,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"g","location":{"start":234,"end":235,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":238,"end":241,"filename":"comparison.rinha"}},"op":"Gte","rhs":{"kind":"Int","value":6,"location":{"start":245,"end":246,"filename":"comparison.rinha"}},"location":{"start":238,"end":246,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":252,"end":253,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"g","location":{"start":262,"end":263,"filename":"comparison.rinha"}},"location":{"start":256,"end":264,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"h","location":{"start":270,"end":271,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":274,"end":277,"filename":"comparison.rinha"}},"op":"Lte","rhs":{"kind":"Int","value":5,"location":{"start":281,"end":282,"filename":"comparison.rinha"}},"location":{"start":274,"end":282,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":288,"end":289,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"h","location":{"start":298,"end":299,"filename":"comparison.rinha"}},"location":{"start":292,"end":300,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"i","location":{"start":306,"end":307,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":310,"end":313,"filename":"comparison.rinha"}},"op":"Lte","rhs":{"kind":"Int","value":4,"location":{"start":317,"end":318,"filename":"comparison.rinha"}},"location":{"start":310,"end":318,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":324,"end":325,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"i","location":{"start":334,"end":335,"filename":"comparison.rinha"}},"location":{"start":328,"end":336,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"tt","location":{"start":342,"end":344,"filename":"comparison.rinha"}},"value":{"kind":"Bool","value":true,"location":{"start":347,"end":351,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"ff","location":{"start":357,"end":359,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"num","location":{"start":362,"end":365,"filename":"comparison.rinha"}},"op":"Neq","rhs":{"kind":"Int","value":5,"location":{"start":369,"end":370,"filename":"comparison.rinha"}},"location":{"start":362,"end":370,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"j","location":{"start":376,"end":377,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"tt","location":{"start":380,"end":382,"filename":"comparison.rinha"}},"op":"And","rhs":{"kind":"Var","text":"tt","location":{"start":386,"end":388,"filename":"comparison.rinha"}},"location":{"start":380,"end":388,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":394,"end":395,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"j","location":{"start":404,"end":405,"filename":"comparison.rinha"}},"location":{"start":398,"end":406,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"k","location":{"start":412,"end":413,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"tt","location":{"start":416,"end":418,"filename":"comparison.rinha"}},"op":"And","rhs":{"kind":"Var","text":"ff","location":{"start":422,"end":424,"filename":"comparison.rinha"}},"location":{"start":416,"end":424,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":430,"end":431,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"k","location":{"start":440,"end":441,"filename":"comparison.rinha"}},"location":{"start":434,"end":442,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"l","location":{"start":448,"end":449,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"ff","location":{"start":452,"end":454,"filename":"comparison.rinha"}},"op":"And","rhs":{"kind":"Var","text":"ff","location":{"start":458,"end":460,"filename":"comparison.rinha"}},"location":{"start":452,"end":460,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":466,"end":467,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"l","location":{"start":476,"end":477,"filename":"comparison.rinha"}},"location":{"start":470,"end":478,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"m","location":{"start":484,"end":485,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"tt","location":{"start":488,"end":490,"filename":"comparison.rinha"}},"op":"Or","rhs":{"kind":"Var","text":"tt","location":{"start":494,"end":496,"filename":"comparison.rinha"}},"location":{"start":488,"end":496,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":502,"end":503,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"m","location":{"start":512,"end":513,"filename":"comparison.rinha"}},"location":{"start":506,"end":514,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"n","location":{"start":520,"end":521,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"tt","location":{"start":524,"end":526,"filename":"comparison.rinha"}},"op":"Or","rhs":{"kind":"Var","text":"ff","location":{"start":530,"end":532,"filename":"comparison.rinha"}},"location":{"start":524,"end":532,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"_","location":{"start":538,"end":539,"filename":"comparison.rinha"}},"value":{"kind":"Print","value":{"kind":"Var","text":"n","location":{"start":548,"end":549,"filename":"comparison.rinha"}},"location":{"start":542,"end":550,"filename":"comparison.rinha"}},"next":{"kind":"Let","name":{"text":"o","location":{"start":556,"end":557,"filename":"comparison.rinha"}},"value":{"kind":"Binary","lhs":{"kind":"Var","text":"ff","location":{"start":560,"end":562,"filename":"comparison.rinha"}},"op":"Or","rhs":{"kind":"Var","text":"ff","location":{"start":566,"end":568,"filename":"comparison.rinha"}},"location":{"start":560,"end":568,"filename":"comparison.rinha"}},"next":{"kind":"Print","value":{"kind":"Var","text":"o","location":{"start":576,"end":577,"filename":"comparison.rinha"}},"location":{"start":570,"end":578,"filename":"comparison.rinha"}},"location":{"start":552,"end":578,"filename":"comparison.rinha"}},"location":{"start":534,"end":578,"filename":"comparison.rinha"}},"location":{"start":516,"end":578,"filename":"comparison.rinha"}},"location":{"start":498,"end":578,"filename":"comparison.rinha"}},"location":{"start":480,"end":578,"filename":"comparison.rinha"}},"location":{"start":462,"end":578,"filename":"comparison.rinha"}},"location":{"start":444,"end":578,"filename":"comparison.rinha"}},"location":{"start":426,"end":578,"filename":"comparison.rinha"}},"location":{"start":408,"end":578,"filename":"comparison.rinha"}},"location":{"start":390,"end":578,"filename":"comparison.rinha"}},"location":{"start":372,"end":578,"filename":"comparison.rinha"}},"location":{"start":353,"end":578,"filename":"comparison.rinha"}},"location":{"start":338,"end":578,"filename":"comparison.rinha"}},"location":{"start":320,"end":578,"filename":"comparison.rinha"}},"location":{"start":302,"end":578,"filename":"comparison.rinha"}},"location":{"start":284,"end":578,"filename":"comparison.rinha"}},"location":{"start":266,"end":578,"filename":"comparison.rinha"}},"location":{"start":248,"end":578,"filename":"comparison.rinha"}},"location":{"start":230,"end":578,"filename":"comparison.rinha"}},"location":{"start":212,"end":578,"filename":"comparison.rinha"}},"location":{"start":194,"end":578,"filename":"comparison.rinha"}},"location":{"start":176,"end":578,"filename":"comparison.rinha"}},"location":{"start":158,"end":578,"filename":"comparison.rinha"}},"location":{"start":140,"end":578,"filename":"comparison.rinha"}},"location":{"start":121,"end":578,"filename":"comparison.rinha"}},"location":{"start":103,"end":578,"filename":"comparison.rinha"}},"location":{"start":85,"end":578,"filename":"comparison.rinha"}},"location":{"start":67,"end":578,"filename":"comparison.rinha"}},"location":{"start":49,"end":578,"filename":"comparison.rinha"}},"location":{"start":31,"end":578,"filename":"comparison.rinha"}},"location":{"start":13,"end":578,"filename":"comparison.rinha"}},"location":{"start":0,"end":578,"filename":"comparison.rinha"}},"location":{"start":0,"end":578,"filename":"comparison.rinha"}}
