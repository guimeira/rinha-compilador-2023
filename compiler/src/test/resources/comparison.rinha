let num = 5;
let b = num > 10;
let _ = print(b);
let b = 10 > num;
let _ = print(b);
let c = num < 10;
let _ = print(c);
let d = num == 10;
let _ = print(d);
let e = num == 5;
let _ = print(e);
let f = num >= 5;
let _ = print(f);
let g = num >= 6;
let _ = print(g);
let h = num <= 5;
let _ = print(h);
let i = num <= 4;
let _ = print(i);
let tt = true;
let ff = num != 5;
let j = tt && tt;
let _ = print(j);
let k = tt && ff;
let _ = print(k);
let l = ff && ff;
let _ = print(l);
let m = tt || tt;
let _ = print(m);
let n = tt || ff;
let _ = print(n);
let o = ff || ff;
print(o)
