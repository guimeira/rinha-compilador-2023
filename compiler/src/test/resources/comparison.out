false
true
true
false
true
true
false
true
false
true
false
false
true
true
false

